### ISSUE TITLE
Secondary point symbol

### ISSUE THREAD

**Issue by ollesmaps** [OpenOrienteering/mapper#1951]
### Steps to reproduce
1. Export from Ocad MTBO symbol set - symbol Forbidden route
2. Import into OOM
3. Skipped secondary point 

### Actual behaviour
Secondary point missing. See zipped att.

### Expected behaviour
Correct import

### Configuration
Mapper Version: 0.9.5
Operating System: Win

![forbidden](https://user-images.githubusercontent.com/20535899/117962369-71d3dc00-b31f-11eb-8dba-0b9b78296338.jpg)
[forbidden-route.zip](https://github.com/OpenOrienteering/mapper/files/6465601/forbidden-route.zip)

**Comment by lpechacek** [comment]
Confirmed. Mapper issues a warning on import ("In line symbol 711.0 'Forbidden route (automatically)': Skipped secondary point symbol.") but I don't see why the import routine skips the secondary symbol.

**Comment by lpechacek** [comment]
Minimal reproducer created with OCAD 8. There should be dots and squares alternating on the line.
[ocad8-minimal-reproducer.zip](https://github.com/OpenOrienteering/mapper/files/6478197/ocad8-minimal-reproducer.zip)

**Comment by dl3sdo** [comment]
![fobidde-route](https://user-images.githubusercontent.com/14225666/118262850-c610c480-b4b5-11eb-8170-1f14c2770678.jpg)
Output with OCAD 12

**Comment by dl3sdo** [comment]
Making progress....
The 	else if (attributes.secondary_data_size > 0) part has a number of issues (of which the first one is the 'else' of course).
Removing the line_symbol->mid_symbol = new OcdImportedPointSymbol(); and replacing the 'elements' parameter in the line below by reinterpret_cast<const Ocd::PointSymbolElementV8*>(coords) leads to something like this:
![fobidde-route_mod](https://user-images.githubusercontent.com/14225666/118285175-7a6c1400-b4d1-11eb-82e6-a31836cba66d.jpg)

**Comment by dl3sdo** [comment]
I manually modified the symbol in Mapper (see attachment). However, adding the 'falling' rows with an offset leads to adding an offset to the end symbol and thus will lead to a line getting longer as configured. I don't see an easy way to import these kind of OCAD symbols as OCAD treats the secondary point symbol in combination with the main one, whereas Mapper processes all line parts forming an element (e.g., the mid element) independently.
BTW: you will experience a weird behaviour if you keep the checkmark at 'Secondary symbol B' while setting the number of Main symbol A to 0 in OCAD.

[forbidden-route_mod.zip](https://github.com/OpenOrienteering/mapper/files/6480553/forbidden-route_mod.zip)

**Comment by dg0yt** [comment]
Thanks @ollesmaps for reporting. "Secondary symbols" do not have a natural counterpart in Mapper, and until now they didn't seem to be necessary. So adding the is not "low-hanging fruit".

As done with some OCD symbol configuratios, this particular use case could be mapped to a combined symbols. This won't be fully accurate, but it might be good enough. The more difficult part is the export to OCD, with rule-based mapping back to a secondary symbol.

In any case, the symbol has some quirks depending on the actual line object., in particular for short lines and for curves. Is this expected or acceptable?

![demo](https://user-images.githubusercontent.com/13567791/118387447-e3ad6d80-b61e-11eb-846e-8021627a5cd3.png)

**Comment by dl3sdo** [comment]
Stretching, shrinking or bending the object length will in most cases lead to gaps or overlaps, which would be ok for me,
This is how it looks in OCAD:
![forbidden-route_OCAD12](https://user-images.githubusercontent.com/14225666/118468523-25184880-b705-11eb-9d87-7b6884864c78.jpg)

**Comment by dg0yt** [comment]
> This is how it looks in OCAD

Is this the line symbol from the OCD file for the original post, i.e. without conversion to and from Mapper?

**Comment by dl3sdo** [comment]
Yes, it is the original symbol and it was done in the original OCAD file (migrated from ocd10 to ocd12 when opening), I added the brown curve to show the underlying bend (just as you did).

**Comment by lpechacek** [comment]
Reassigning to Kai who has a working fix. All my example symbols render correctly with the changes from PR #1958.

**Comment by dl3sdo** [comment]
@lpechacek : can you test the fix with the attached ocd files? Both are somehow 'invalid' as I set the number of Main symbol A to 0, so they will only contain the secondary symbol.
[fobidde-route_ZeroMainA.zip](https://github.com/OpenOrienteering/mapper/files/6515921/fobidde-route_ZeroMainA.zip)

**Comment by lpechacek** [comment]
This is the outcome of the zero number of main symbols (fobidde-route_ZeroMainA.zip):

![zero-main-a](https://user-images.githubusercontent.com/5584406/119119600-9fa9d680-ba2b-11eb-8ea2-c053eecb2303.png)

**Comment by lpechacek** [comment]
The previous test motivated me to try multiple main symbols on the line. I never experimented with this setting. At the top, it's OCAD rendering. At the bottom is the Mapper variant. The main symbol is a perpendicular dash, repeated four times. The squares are the secondary symbol.

![multi-main-symbol](https://user-images.githubusercontent.com/5584406/119128558-18ae2b80-ba36-11eb-9504-c7005f489cbd.png)

As seen in the picture, OCAD does not bother squeezing the main symbol into the line footprint. It would be understandable if the interpretation is that the main symbol is centered on the line ends, but the rendering differs at the start and end of line. I'm not sure if we should poke more into this bug nest.

**Comment by dg0yt** [comment]
I don't want to deal to much with the undefined behaviours ("somehow 'invalid' ... set the number of Main symbol A to 0") while still dreaming of LineSymbol 2.0 in Mapper. (Well, I somewhat use CombinedSymbol in that way :smile:)
However, I would appreciate some robustness around the use cases we want to support, such as that MTBO symbol. In particular, I did not yet tests with any dash style options.
